doc/source/rllib/rllib-concepts.rst

```diff
@@ -230,76 +230,60 @@ First, check out the `PPO trainer definition <https://github.com/ray-project/ray
-PPO can switch between a simple synchronous plan, or a multi-GPU plan that implements minibatch SGD (the default):
+PPO can switch between a simple synchronous optimizer, or a multi-GPU plan that implements minibatch SGD (the default):
 
 .. code-block:: python
 
-    def execution_plan(workers: WorkerSet, config: TrainerConfigDict):
-        rollouts = ParallelRollouts(workers, mode="bulk_sync")
-
-        # Collect large batches of relevant experiences & standardize.
-        rollouts = rollouts.for_each(
-            SelectExperiences(workers.trainable_policies()))
-        rollouts = rollouts.combine(
-            ConcatBatches(min_batch_size=config["train_batch_size"]))
-        rollouts = rollouts.for_each(StandardizeFields(["advantages"]))
-
-        if config["simple_optimizer"]:
-            train_op = rollouts.for_each(
-                TrainOneStep(
-                    workers,
-                    num_sgd_iter=config["num_sgd_iter"],
-                    sgd_minibatch_size=config["sgd_minibatch_size"]))
+        def training_iteration(self) -> ResultDict:
+        # Collect SampleBatches from sample workers until we have a full batch.
+        if self._by_agent_steps:
+            train_batch = synchronous_parallel_sample(
+                worker_set=self.workers, max_agent_steps=self.config["train_batch_size"]
+            )
         else:
-            train_op = rollouts.for_each(
-                TrainTFMultiGPU(
-                    workers,
-                    sgd_minibatch_size=config["sgd_minibatch_size"],
-                    num_sgd_iter=config["num_sgd_iter"],
-                    num_gpus=config["num_gpus"],
-                    rollout_fragment_length=config["rollout_fragment_length"],
-                    num_envs_per_worker=config["num_envs_per_worker"],
-                    train_batch_size=config["train_batch_size"],
-                    shuffle_sequences=config["shuffle_sequences"],
-                    _fake_gpus=config["_fake_gpus"]))
-
-        # Update KL after each round of training.
-        train_op = train_op.for_each(lambda t: t[1]).for_each(UpdateKL(workers))
-
-        return StandardMetricsReporting(train_op, workers, config) \
-            .for_each(lambda result: warn_about_bad_reward_scales(config, result))
-
-Suppose we want to customize PPO to use an asynchronous-gradient optimization strategy similar to A3C. To do that, we could swap out its execution plan to that of A3C's:
-
-.. code-block:: python
-
-    from ray.rllib.agents.ppo import PPOTrainer
-    from ray.rllib.execution.rollout_ops import AsyncGradients
-    from ray.rllib.execution.train_ops import ApplyGradients
-    from ray.rllib.execution.metric_ops import StandardMetricsReporting
-
-    def a3c_execution_plan(workers, config):
-        # For A3C, compute policy gradients remotely on the rollout workers.
-        grads = AsyncGradients(workers)
+            train_batch = synchronous_parallel_sample(
```

Comment:
Again, I think we should begin referencing our own in-library code wherever possible.
We should refrain from keeping possibly outdated copies of all these things around. That is really hard to properly maintain in the long term.

Reply:
^ @maxpumperla Can you maybe draw a line where it makes sense to write a separate example and where it makes sense to reference our in-library code?

Reply:
I agree, we should:
* reference in-library code as much as possible
* if we are discussing an example (no in-library code): move the actual code from the docs into examples/documentation/... and link to that from the docs instead. Also make sure the examples/documentation/... code is tested via BUILD